tests/cases/compiler/jsExportMemberMergedWithModuleAugmentation.ts

```diff
@@ -0,0 +1,27 @@
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+
+// @Filename: /test.js
+class Abcde {
+  /** @type {string} */
+  x;
+}
+
+module.exports = {
+  Abcde
+};
+
+// @Filename: /index.ts
+import { Abcde } from "./test";
+
+declare module "./test" {
+  interface Abcde { b: string }
+}
+
+new Abcde().x;
+
+// Descriptive, not prescriptive: This is ok,
+// because the type meaning from /test.js does
+// not propagate through the object literal export.
```

Comment:
This seems like a bug in our implementation that should be fixed someday, at which point the assignment won't be allowed. I think "currently buggy" is a better description than "Descriptive, not prescriptive" in that case. But, I may have misunderstood your intent.